### ISSUE TITLE
Redirect debug output in vscode on linux

### ISSUE THREAD

**Issue by marcsosduma** [OlegKunitsyn/gnucobol-debug#132]
Hey!

I'm using the gnucobol-debug to debug programs on linux. These programs use formatted screens in the 'screen section'. To output the program in a separate vscode window, I'm using two scripts that I've placed in /usr/local/bin:

newterm.sh
`xterm -fa 'Monospace' -fs 16 -e '/usr/bin/tty >~/.numterm; sleep 1000000;'`

windb.sh
```
newterm.sh &
sed -i 's/set.*inferior.*//g' ~/.gdbinit
sed -i '/^$/d' ~/.gdbinit
sleep 1
term=`cat ~/.numterm`
echo $term
echo "set inferior-tty $term" >> ~/.gdbinit
```

Before running the debug, I run "windb.sh" which opens an xterm window to receive the output of my program.

Is this the best way?

Thanks!

**Comment by GitMensch** [comment]
That's really useful! I'd just suggest that you do all this in the launch.json - execute external task in `preLaunchTask` and instead of tinkering with the gdbinit file it would be best to be able to have `autorun` or similar (commands that will be sent to gdb after startup before launching the process) - but sadly that's not available yet.
@OlegKunitsyn: Would you mind creating a FR and/or (better) implement it as a string array option?

I'd personally prefer to still do that within gdb to do so you can execute these commands from within GDB - see https://stackoverflow.com/questions/6885923/redirecting-storing-output-of-shell-into-gdb-variable for any example.

**Comment by marcsosduma** [comment]
Hello!
This method only works correctly when I run the vscode (or vscodium) and windb.sh from the terminal:
```
>cd /myproject
>windb.sh
>code .
```
Do you know why?

**Comment by marcsosduma** [comment]
Hey!

Trying to follow your suggestion I got to this point:

.gdbinit
`source external_debug.py`

external_debug.py
```
import os
from pathlib import Path
import subprocess, signal
import time
import platform
import gdb

def read_num_terminal():
    home = Path.home()
    fileName = os.path.join(home, ".numterm")
    if not os.path.isfile(fileName):
        return None
    file1 = open(fileName, "r")
    termName = file1.readlines()[0].strip();
    file1.close()
    return termName

def isTermOn():
    bTermIsOn = False
    termName= read_num_terminal()
    #file exists?
    if termName:
        searchStr= termName.split("/")[2]+"/"+termName.split("/")[3]
        ps = subprocess.Popen(('ps', '-e'), stdout=subprocess.PIPE)
        output = ps.communicate()[0]
        for line in output.decode("utf8").split('\n'):
            #print(line)
            if line.find(searchStr)>=0 and line.find("sleep"):
                bTermIsOn=True
    return bTermIsOn

def main():
    command1 = None
    if platform.system() == "Linux":
        # Open the new terminal - xterm
        if not isTermOn():
            process = subprocess.Popen(["xterm","-title","GnuCobol Debug","-fa",
                                                           "DejaVu Sans Mono", "-fs", "12", "-e",
                                                           "/usr/bin/tty>~/.numterm; sleep 1000000;"],
                            stdin =subprocess.DEVNULL,
                            stdout=subprocess.PIPE, 
                            stderr=subprocess.PIPE,
                            start_new_session=True,
                            preexec_fn=(lambda: signal.signal(signal.SIGHUP, signal.SIG_IGN))
                            )
            # Read the tty device name
            time.sleep(1)
        # Redirect to terminal
        termName= read_num_terminal()
        # mi or mi2?
        command1 = 'interpreter-exec mi "-inferior-tty-set ' + termName +'"' # Using readlines()
    else:
        command1 = 'interpreter-exec mi "set new-console"'
    print(command1)
    gdb.execute(command1)
 

if __name__ == "__main__":
    main()

```

To use, as in the previous way, I need to run the vscode (or vscodium) from the terminal:

cd /myproject
code .

and then "F5" to start the debug.

I couldn't get it to work on windows...
Any suggestion?

**Comment by OlegKunitsyn** [comment]
> Would you mind creating a FR and/or (better) implement it as a string array option?
Since both options lead to the code, a MR might be sufficient.

**Comment by marcsosduma** [comment]
Hello! Oleg.
Congratulations on the Gnucobol-Debug Extension.
I don't really know how to do what you're asking, but I'll look into it.

**Comment by marcsosduma** [comment]
Hello! Oleg and Simon,

as you indicated, I changed the code to allow debugging in a separate window. I did the tests and it works on Windows and Linux.

Thanks,

**Comment by OlegKunitsyn** [comment]
Thank you very much @marcsosduma 
Can you fix a typo and squash your commits please?

**Comment by marcsosduma** [comment]
Hello! Please check that everything is OK. Thanks.

**Comment by GitMensch** [comment]
Closed by #133.
